File: hw/viteri/u3/HW30ObserverPatternandStrategy/HW30Strategy/src/main/java/model/BubbleSort.java
/*
 * Click nbfs://nbhost/SystemFileSystem/Templates/Licenses/license-default.txt to change this license
 * Click nbfs://nbhost/SystemFileSystem/Templates/Classes/Class.java to edit this template
 */
package model;

import controller.SortingStrategy;

/**
 *
 * @author Alexis Viteri DCO-ESPE
 */
public class BubbleSort extends SortingStrategy {

    @Override
    public void sort(int[] array) {
        for (int reverseIndex = array.length - 1; reverseIndex > 1; reverseIndex--) {

            for (int i = 0; i < reverseIndex; i++) {

                if (array[i] > array[i + 1]) {
                    int temp = array[i];

                    array[i] = array[i + 1];
                    array[i + 1] = temp;

                }
            }
        }

    }

}

File: exams/Vaca/U3/ExamUnit3/model/BubbleSort.js
const SortingStrategy = require('./SortingStrategy');

class BubbleSort extends SortingStrategy {
    sort(array) {
        // Implementación del algoritmo BubbleSort
        for (let i = 0; i < array.length - 1; i++) {
            for (let j = 0; j < array.length - 1 - i; j++) {
                if (array[j] > array[j + 1]) {
                    [array[j], array[j + 1]] = [array[j + 1], array[j]];
                }
            }
        }
        return array;
    }
}

module.exports = BubbleSort;
